auction/src/lib.rs

```diff
@@ -610,35 +670,49 @@ impl<T: Config> Pallet<T> {
 		new_owner: &T::AccountId,
 		price: BalanceOf<T>,
 		balance_source: Option<T::AccountId>,
-	) -> DispatchResult {
-		// Handle marketplace fees
+	) -> Result<(BalanceOf<T>, BalanceOf<T>), DispatchError> {
+		// Handle marketplace fees.
 		let marketplace = T::MarketplaceExt::get_marketplace(auction.marketplace_id)
 			.ok_or(Error::<T>::MarketplaceNotFound)?;
+		let mut nft = T::NFTExt::get_nft(nft_id).ok_or(Error::<T>::NFTNotFound)?;
 
-		let to_marketplace =
-			price.saturating_mul(marketplace.commission_fee.into()) / 100u32.into();
-		let to_auctioneer = price.saturating_sub(to_marketplace);
+		let mut commission_fee_amount: BalanceOf<T> = 0u32.into();
+		if let Some(commission_fee) = &marketplace.commission_fee {
```

Comment:
I suggest you to implement a method for `CompoundFee` that is going to return the correct commission fee amount.
Example: 

```rust
get_flat_fee(price: Balance): Balance {
  match *self {
    Self::Flat(x) =>  x,
    Self::Percentage(x) => x * price,
  };
}
```

Then you can do this kind of calls:
```rust
let commission_fee_amount = marketplace.commission_fee.map_or_else(|| 0, |x| x.get_flat_fee());
```